packages/query-core/src/queryClient.ts

```diff
@@ -511,18 +511,15 @@ export class QueryClient {
     }
 
     // dependent default values
-    if (typeof defaultedOptions.refetchOnReconnect === 'undefined') {
+    if (defaultedOptions.refetchOnReconnect === undefined) {
       defaultedOptions.refetchOnReconnect =
         defaultedOptions.networkMode !== 'always'
     }
-    if (typeof defaultedOptions.throwOnError === 'undefined') {
+    if (defaultedOptions.throwOnError === undefined) {
       defaultedOptions.throwOnError = !!defaultedOptions.suspense
     }
 
-    if (
-      typeof defaultedOptions.networkMode === 'undefined' &&
-      defaultedOptions.persister
-    ) {
+    if (!defaultedOptions.networkMode && defaultedOptions.persister) {
```

Comment:
`networkMode` is a string literal type so doing the simplified `!` check should be safe. We can't do that for the above booleans obviously